### ISSUE TITLE
Particle creation diagnostics?

### ISSUE THREAD

**Issue by lucafedeli88** [ECP-WarpX/WarpX#1664]
Hi!
I was thinking that in some simulations it could be interesting to store particle creation events, in order to answer questions like:  "where did photon emission occur?" "what were the properties of positron particles when they were created?"  " what was the local EM field when a ionization event happened?"
Now these pieces of information can be obtained only if particle data is dumped at every time step and it requires very expensive post processing.
Do you think that this diagnostic could be useful in WarpX?

**Comment by NeilZaim** [comment]
Yes, I think such a diag would be quite useful for (at least) QED simulations :) .

I guess one way to achieve this could be to have the possibility to store extra particle variables in the SoA (e.g. the initial particle position). This would be useful for more things than just particle creation but I guess quite bad in terms of memory usage.

Or did you have in mind a diagnostic more specific to particle creation (e.g. that would store particle creation events in a buffer that is dumped periodically)?

**Comment by lucafedeli88** [comment]
I had in mind something like the second solution that you have mentioned: storing creation events in a buffer and dump it every few timesteps or when it is full. However, if we decide to implement this diagnostic we can consider in detail all the possible options...
